### PR TITLE
replace rust toolchain build action with maintained one

### LINKED ISSUE #833: Replace actions-rs/toolchain with rustup-toolchain-install
Per [@wilwade's comment](https://github.com/LibertyDSNP/frequency/pull/828#discussion_r1054375892),  we should switch off of `actions-rs/toolchain` action to https://github.com/marketplace/actions/rustup-toolchain-install. It is actually maintained and looks like it is the go to replacement.

```rust
uses: actions-rs/toolchain@16499b5e05bf2e26879000db0c1d13f7e13fa3af
```

**Tasks**:

- [x] Replace with new action in Verify PR workflow
- [x] Add new action to the authorized list in [Organization Settings](https://github.com/organizations/LibertyDSNP/settings/actions)
- [x] Collect feedback from @wilwade and @saraswatpuneet 
- [x] Replace with new action in the remaining workflows
- [ ] Delete test RS file
- [ ] Clean. up after test release

### PR DESCRIPTION
**Pull request by demisx**
# Goal
The goal of this PR is to replace Rust toolchain GitHub action with the one that's currently maintained.

Closes #833

### CODE REVIEW

File: .github/workflows/verify-pr-commit.yml [hunk 1]
```diff
@@ -125,24 +125,22 @@ jobs:
...
+          path: ${{env.BIN_DIR}}/${{env.FINAL_BIN_FILENAME}}
+          key: binaries-${{runner.os}}-${{env.NETWORK}}-${{github.head_ref}}
       - name: Install Required Packages
         run: |
           sudo apt-get update
           sudo apt install -y protobuf-compiler
       - name: Install Rust Toolchain
         if: steps.cache-binary.outputs.cache-hit != 'true'
-        uses: actions-rs/toolchain@16499b5e05bf2e26879000db0c1d13f7e13fa3af
+        uses: dtolnay/rust-toolchain@e12eda571dc9a5ee5d58eecf4738ec291c66f295
```
[the remark this answers is not in the document]
Comment (wilwade) [reply .github/workflows/verify-pr-commit.yml:141]: That's fine.

File: .github/workflows/verify-pr-commit.yml [hunk 2]
```diff
@@ -102,7 +102,7 @@ jobs:
             branch_alias: pr
     runs-on: [self-hosted, Linux, X64]
     env:
-      HOME: /home/runner
+      HOME: /root
```
[the remark this answers is not in the document]
Comment (wilwade) [reply .github/workflows/verify-pr-commit.yml:105]: @demisx Will this break if we go back to free runners?

File: .github/workflows/verify-pr-commit.yml [hunk 3]
```diff
@@ -125,24 +125,22 @@ jobs:
...
       - name: Install Rust Toolchain
         if: steps.cache-binary.outputs.cache-hit != 'true'
-        uses: actions-rs/toolchain@16499b5e05bf2e26879000db0c1d13f7e13fa3af
+        uses: dtolnay/rust-toolchain@e12eda571dc9a5ee5d58eecf4738ec291c66f295
         with:
-          default: true
-          profile: minimal
-          target: wasm32-unknown-unknown
+          targets: wasm32-unknown-unknown
           toolchain: stable
```
[the remark this answers is not in the document]
Comment (wilwade) [reply .github/workflows/verify-pr-commit.yml:144]: I think it is ok, but @saraswatpuneet might know differently.

[the remark this answers is not in the document]
Comment (saraswatpuneet) [reply .github/workflows/verify-pr-commit.yml:144]: Should be fine we override it with toolchain file

**Review by wilwade — APPROVED** [review]

**Review by saraswatpuneet — APPROVED** [review]

**Pull request merged**